vmware-event-router/internal/provider/vcsim/vcsim_test.go

```diff
@@ -0,0 +1,153 @@
+// +build unit
+
+package vcsim
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"reflect"
+	"testing"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/pkg/errors"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/metrics"
+	"github.com/vmware-samples/vcenter-event-broker-appliance/vmware-event-router/internal/processor"
+)
+
+func Test_reverse(t *testing.T) {
+	vmOnEvent := types.VmPoweredOnEvent{}
+	vmOffEvent := types.VmPoweredOffEvent{}
+
+	type args struct {
+		events []types.BaseEvent
+	}
+	tests := []struct {
+		name string
+		args args
+		want []types.BaseEvent
+	}{
+		{
+			name: "empty slice",
+			args: args{
+				events: nil,
+			},
+			want: nil,
+		},
+		{
+			name: "two entries",
+			args: args{
+				events: []types.BaseEvent{&vmOnEvent, &vmOffEvent},
+			},
+			want: []types.BaseEvent{&vmOffEvent, &vmOnEvent},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			reverse(tt.args.events)
+			if !reflect.DeepEqual(tt.args.events, tt.want) {
+				t.Errorf("reverse() gotReverse = %v, want %v", tt.args.events, tt.want)
+			}
+		})
+	}
+}
+
+func Test_eventHandler(t *testing.T) {
```

Comment:
Good tests!

Reply:
Thank you :) You're the baseline for solid tests ;)